parquet/src/main/scala/monix/connect/parquet/ParquetPublisher.scala

```diff
@@ -31,56 +33,56 @@ import scala.util.{Failure, Success}
 
 /**
   * The implementation of a reactive parquet publisher.
-  * @param reader The apache hadoop generic implementation of a parquet reader.
+  * @param parquetReader The apache hadoop generic implementation of a parquet reader.
   */
-private[parquet] final class ParquetPublisher[T](reader: ParquetReader[T]) extends Observable[T] {
+private[parquet] final class ParquetPublisher[T](parquetReader: Task[ParquetReader[T]]) extends Observable[T] {
 
   def unsafeSubscribeFn(subscriber: Subscriber[T]): Cancelable = {
     val s = subscriber.scheduler
     val cancelable = BooleanCancelable()
-    fastLoop(subscriber, cancelable, s.executionModel, 0)(s)
+    Resource.make(parquetReader)(reader => Task(reader.close()))
+      .use(r => fastLoop(subscriber, cancelable, s.executionModel, 0, r)(s))
+      .runToFuture(s)
```

Comment:
@Avasil could you review this new implementation that aims to safely read from parquet files from a given `Task[ParquetWriter]`?

Reply:
I think it should be the other way around.
`def reader` should use `fromReaderUnsafe` in the implementation.
For instance, `Observable.resource(readerTask)(reader => Task(reader.close())).flatMap(fromReaderUnsafe)`

If we pass existing `ParquetReader` and not a "factory" then it's a responsibility of a user (producer) to close it
If we have `Task[ParquetReader]` then we will have a generator and `Observable` will be the producer so we should close it

Reply:
Ahh, I also thought about doing that, but are we responsible of closing user's resources? They might want to use it somewhere else and when they do, it will throw an exception. 
Just trying to clarify myself with the purpose of doing that, don't mind rolling the implementation back at all ;)

Reply:
If they pass `reader: Task[ParquetReader]` they can't do that unless they cheat 

Each time we use this `Task`, we create a new reader that the user doesn't have access to

In case of `reader: ParquetReader` then yes, they might want to use it somewhere else so we don't close it